services/explorer-indexer/src/modules/historical-sync/historical.syncer.service.ts

```diff
@@ -192,48 +217,59 @@ export default class HistoricalSyncService implements OnModuleInit {
     // @note : Right now there is no way to get the genesis transactions to set:
     // Initial validators, initial balances for delegators and faucet.
     // big TODO
-    const chuncks = [
+    const chunks = [
       {
         name: QueueJobs.NEW_PBFT_BLOCKS,
         data: {
           pbftPeriod: 0,
         },
       },
     ];
-    const foundBlockNumbers = await this.pbftService.getSavedPbftPeriods();
-    const maxVal = foundBlockNumbers.sort((a, b) => b - a)[0];
-
-    for (let i = 1; i <= maxVal; i++) {
-      if (foundBlockNumbers.indexOf(i) == -1) {
-        chuncks.push({
-          name: QueueJobs.NEW_PBFT_BLOCKS,
-          data: {
-            pbftPeriod: i,
-            type: SyncTypes.HISTORICAL,
-          } as QueueData,
-        });
+    const missingBlockNumbers = await this.pbftService.getMissingPbftPeriods();
```

Comment:
And now, instead of getting the ones saved and then checking them and not including those, we simply iterate through the ones we don't have a push them in with the same chunky approach.